linux/Platform.c

```diff
@@ -430,117 +432,91 @@ char* Platform_getProcessEnv(pid_t pid) {
    return env;
 }
 
-/*
- * Return the absolute path of a file given its pid&inode number
- *
- * Based on implementation of lslocks from util-linux:
- * https://sources.debian.org/src/util-linux/2.36-3/misc-utils/lslocks.c/#L162
- */
-char* Platform_getInodeFilename(pid_t pid, ino_t inode) {
-   struct stat sb;
+FileLocks_ProcessData* Platform_getProcessLocks(pid_t pid) {
+   FileLocks_ProcessData* pdata = xCalloc(1, sizeof(FileLocks_ProcessData));
    const struct dirent* de;
    DIR* dirp;
-   ssize_t len;
-   int fd;
+   int dfd;
 
    char path[PATH_MAX];
-   char sym[PATH_MAX];
-   char* ret = NULL;
-
-   memset(path, 0, sizeof(path));
-   memset(sym, 0, sizeof(sym));
-
-   xSnprintf(path, sizeof(path), "%s/%d/fd/", PROCDIR, pid);
+   xSnprintf(path, sizeof(path), PROCDIR "/%d/fdinfo/", pid);
    if (strlen(path) >= (sizeof(path) - 2))
-      return NULL;
+      goto err;
 
    if (!(dirp = opendir(path)))
-      return NULL;
+      goto err;
 
-   if ((fd = dirfd(dirp)) < 0 )
-      goto out;
+   if ((dfd = dirfd(dirp)) == -1) {
+      closedir(dirp);
+      goto err;
+   }
 
+   FileLocks_LockData** data_ref = &pdata->locks;
    while ((de = readdir(dirp))) {
       if (String_eq(de->d_name, ".") || String_eq(de->d_name, ".."))
          continue;
 
-      /* care only for numerical descriptors */
-      if (!strtoull(de->d_name, (char **) NULL, 10))
+      errno = 0;
+      char *end;
```

Comment:
Avoid uninitialized memory access if `strtoull` does not set `end` (DiD):
```suggestion
      char *end = de->d_name;
```

Reply:
Doing just this causes the build failure.

Please add this patch to resolve the build issue:
```diff
diff --git a/linux/Platform.c b/linux/Platform.c
index f5aa6b96..64f25c42 100644
--- a/linux/Platform.c
+++ b/linux/Platform.c
@@ -434,7 +434,6 @@ char* Platform_getProcessEnv(pid_t pid) {
 
 FileLocks_ProcessData* Platform_getProcessLocks(pid_t pid) {
    FileLocks_ProcessData* pdata = xCalloc(1, sizeof(FileLocks_ProcessData));
-   const struct dirent* de;
    DIR* dirp;
    int dfd;
 
@@ -452,7 +451,7 @@ FileLocks_ProcessData* Platform_getProcessLocks(pid_t pid) {
    }
 
    FileLocks_LockData** data_ref = &pdata->locks;
-   while ((de = readdir(dirp))) {
+   for (struct dirent* de; (de = readdir(dirp)); ) {
       if (String_eq(de->d_name, ".") || String_eq(de->d_name, ".."))
          continue;
 
```